src/renderer/helpers/api/invidious.js

```diff
@@ -142,6 +266,118 @@ export async function invidiousGetCommentReplies({ id, replyToken }) {
   return { commentData: parseInvidiousCommentData(response), continuation: response.continuation ?? null }
 }
 
+/**
+ * @param {string} query
+ * @returns {Promise<string[]>}
+ */
+export async function getInvidiousSearchSuggestions(query) {
+  return await invidiousAPICall({
+    resource: 'search/suggestions',
+    id: '',
+    params: {
+      q: query
+    }
+  })
+}
+
+/**
+ * @returns {Promise<any[]>}
+ */
+export async function getInvidiousPopularFeed() {
+  const response = await invidiousAPICall({
+    resource: 'popular',
+    id: '',
+    params: {}
+  })
+
+  const items = response.filter((item) => {
+    return item.type === 'video' || item.type === 'shortVideo' || item.type === 'channel' || item.type === 'playlist'
+  })
+
+  items.forEach((item) => {
+    if (item.type === 'video' || item.type === 'shortVideo') {
+      setPublishedTimestamp(item)
+    }
+  })
+
+  return items
+}
+
+/**
+ * @param {'default' | 'music' | 'gaming' | 'movies'} tab
+ * @param {string} region
+ * @returns {Promise<any[] | null>}
+ */
+export async function getInvidiousTrending(tab, region) {
+  const params = {
+    resource: 'trending',
+    id: '',
+    params: {
+      region
+    }
+  }
+
+  if (tab !== 'default') {
+    params.params.type = tab.charAt(0).toUpperCase() + tab.substring(1)
+  }
+
+  const response = await invidiousAPICall(params)
+
+  if (!response) {
+    return null
+  }
+
+  const items = response.filter((item) => {
+    return item.type === 'video' || item.type === 'channel' || item.type === 'playlist'
+  })
+
+  items.forEach((item) => {
+    if (item.type === 'video') {
+      setPublishedTimestamp(item)
+    }
+  })
+
+  return items
+}
+
+/**
+ * @param {string} query
+ * @param {number} page
+ * @param {any} searchSettings
+ * @returns {Promise<any[] | null>}
+ */
+export async function getInvidiousSearchResults(query, page, searchSettings) {
+  let results = await invidiousAPICall({
+    resource: 'search',
+    id: 'string',
```

Comment:
My reply to @efb4f5ff-1298-471a-8973-3d47447115dc  was about search suggestions, didnt realize there was an issue with search as well whoops.

```suggestion
    id: '',
```

Reply:
Whoops I must have had the cursor in the wrong place while writing one of the JSDoc comments